server/src/main/java/org/elasticsearch/action/search/TransportSearchScrollAction.java

```diff
@@ -40,6 +42,14 @@ public TransportSearchScrollAction(
 
     @Override
     protected void doExecute(Task task, SearchScrollRequest request, ActionListener<SearchResponse> listener) {
+        ActionListener<SearchResponse> loggingListener = listener.delegateFailureAndWrap((l, searchResponse) -> {
+            if (searchResponse.getShardFailures() != null && searchResponse.getShardFailures().length > 0) {
+                for (ShardSearchFailure f : searchResponse.getShardFailures()) {
+                    logger.warn("TransportSearchScrollAction shard failure (partial results response) for request " + request, f);
```

Comment:
Do we want to limit to failures that would lead to >=500 status codes here as well?
Maybe it would also make sense to do some failure deduplication similar to what we do on the rest response rendering side (SearchResponse#headerToXContent() which in turn is using RestActions.buildBroadcastShardsHeader) here at the logging site as well, wdyt?

Reply:
Oh, the `ExceptionsHelper.groupBy(searchResponse.getShardFailures())` will work nicely here. I didn't know about that. Thanks for the pointer.

In terms of checking for 500 status type errors, all of these will be 500 errors since they do not extend Elasticsearch exception. We could potentially check the underlying cause, but then you may have several causes or suppressed exceptions so which to check is unclear.

Reply:
I did end up adding a check for 500 status of the cause (if one is present).